### ISSUE TITLE
Retrieve data from similarity matrix

### ISSUE THREAD

**Issue by voigtjessica** [MaartenGr/BERTopic#926]
Hello Maarten. First of all: thank you for BERTopic, it is an amazing library!
I have a question related to another solved issue. I was looking for a way to retrieve the cosine similarity matrix between topic embeddings and I found this issue [Extracting Similarity/Distance Matrix and Labels #292](https://github.com/MaartenGr/BERTopic/issues/292 ) . I tested the solution proposed by you but the final matrix is not giving me the same numbers the heatmap is giving.

I'm not sure if i'm doing something wrong, but here is my code:

```
  n_neighbors = 20
  min_dist = 0.25
  min_cluster = 30
  min_sample = 7


umap_model  = UMAP(n_neighbors=n_neighbors, 
                            min_dist = min_dist,
                            n_components=5, 
                            metric='cosine', random_state=42)


hdbscan_model = HDBSCAN(min_cluster_size=min_cluster, 
                            min_samples = min_sample, 
                            metric='euclidean', 
                            prediction_data=True)

vectorizer_model = CountVectorizer(stop_words=german_stopwords_alterada) #german_stop_words alterada

topic_model = BERTopic(language = "german", vectorizer_model=vectorizer_model, 
                        umap_model=umap_model, 
                        hdbscan_model=hdbscan_model).fit(docs)

topics, probs = topic_model.fit_transform(docs)

## Creating matrix:

distance_matrix = cosine_similarity(np.array(topic_model.topic_embeddings_)[1:, :])
labels = (topic_model.get_topic_info().sort_values("Topic", ascending=True).Name)[1:]

````

I get a 9x9 matrix (as the result of my heatmap) but all the numbers are different, and the topic similarity among topics in the matrix are lower than the values in the heatmap. Am I doing something wrong? I tried to restart the kennel, but it keep me giving different results. Thank you in advance for any orientation!

**Comment by MaartenGr** [comment]
At face value, I cannot find anything that immediately looks different from the heatmap implementation. Having said that, it might be worthwhile to go through the code [here](https://github.com/MaartenGr/BERTopic/blob/master/bertopic/plotting/_heatmap.py) to see if anything matches. You can actually copy-paste most of the code without needing to do any customization other than changing `return fig`  to `return distance_matrix, new_labels`.

**Comment by voigtjessica** [comment]
Thank you for your answer!
I changed the function as you suggested and got the same results, but now I understand what is happening: the heatmap and matrix are displaying values in different order. So, based on that other issue I'm using the code:

```
sim_matrix = cosine_similarity(np.array(topic_model.topic_embeddings_))
df = pd.DataFrame(sim_matrix, columns=topic_model.topic_labels_.values(), index=topic_model.topic_labels_.values())
```
and while my df shows the value 0.72 for the topics 0x1 , my heatmap is showing the same value for the topics 1x2  and so one. I'm looking in the original ```def heatmap()``` . I see you sort the topics in the function but I'm not sure how to do it for the labels.

**Comment by MaartenGr** [comment]
If you have the topic values, then you could also use `topic_model.topic_labels_ (topic)` to extract and map the labels for the topics that you have.

**Comment by MaartenGr** [comment]
Due to inactivity, I'll be closing this issue. Let me know if you want me to re-open the issue!
